File: Trie/11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    
    let l = 0
    let r = height.length-1
    let res = 0
    while (l<r) {
        if (height[l]<=height[r]) {
            res = Math.max(res, (r-l) * (height[l]))
            l+=1
        } else {
            res = Math.max(res, (r-l) * (height[r]))
            r-=1
        }
    }

    return res
};

File: Sorting/11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0 
        r = len(height)-1
        maxArea = 0

        while l<r:
            maxArea = max(maxArea, (r-l)*min(height[l], height[r]))
            if height[l] < height[r]:
                l+=1
            else:
                r-=1
        return maxArea